kedro/io/core.py

```diff
@@ -374,13 +374,14 @@ def parse_dataset_definition(
         raise DataSetError("`type` is missing from DataSet catalog configuration")
 
     class_obj = config.pop("type")
+    is_tracking_ds = False
     if isinstance(class_obj, str):
         if len(class_obj.strip(".")) != len(class_obj):
             raise DataSetError(
                 "`type` class path does not support relative "
                 "paths or paths ending with a dot."
             )
-
+        is_tracking_ds = class_obj.split(".")[0] == "tracking"
```

Comment:
I think this might be too strict. Technically users can also specify class paths as `kedro.extras.datasets.tracking`, or even have a `tracking` module in their own project, that has nothing to do with our own. Can we check against the `class_obj.__module__` instead?

Reply:
Good point, I'll change it.